client/src/main/java/com/paypal/selion/platform/mobile/android/UiSlider.java

```diff
@@ -0,0 +1,66 @@
+|  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance     |
+|  with the License.                                                                                                  |
+|                                                                                                                     |
+|  You may obtain a copy of the License at                                                                            |
+|                                                                                                                     |
+|       http://www.apache.org/licenses/LICENSE-2.0                                                                    |
+|                                                                                                                     |
+|  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed   |
+|  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for  |
+|  the specific language governing permissions and limitations under the License.                                     |
+\*-------------------------------------------------------------------------------------------------------------------*/
+
+package com.paypal.selion.platform.mobile.android;
+
+import com.paypal.selion.logger.SeLionLogger;
+import com.paypal.selion.platform.grid.SeLionAppiumAndroidDriver;
+import com.paypal.selion.platform.grid.SeLionSelendroidDriver;
+import com.paypal.selion.platform.mobile.elements.MobileSlider;
+import com.paypal.test.utilities.logging.SimpleLogger;
+import org.openqa.selenium.Dimension;
+import org.openqa.selenium.Point;
+import org.openqa.selenium.WebElement;
+
+/**
+ * <code>UiSlider</code> represents a Slider widget for Android UI automation framework.
+ */
+public class UiSlider extends UiObject implements MobileSlider {
+    private static final SimpleLogger logger = SeLionLogger.getLogger();
+
+    public UiSlider(String locator) {
+        super(locator);
+    }
+
+    /**
+     * it is not accurate and is best to used only for setting value to 0 or 1, otherwise the result is close to number
+     * @param value The desired decimal value from 0 to 1, inclusive. A 0 value represents far left and a value of 1
+     */
+    @Override
+    public void dragToValue(double value) {
+        logger.entering(value);
+        WebElement webElement = findElement(locator);
+        Point currentLocation = webElement.getLocation();
+        Dimension elementSize = webElement.getSize();
+        int x = currentLocation.getX();
+        int y = currentLocation.getY() + (elementSize.getHeight() / 2);
+        int pos = Double.valueOf(value * (elementSize.getWidth())).intValue();
+        int endX = x + pos;
+        // full width will throw exception
+        if (value >= 1) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid using Literals in Conditional Statements](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2360069774/issues/source?bid=2928425&fileBranchId=3219780#l52)